### PR TITLE
Replace some usages of LegacyUnit

### PR DESCRIPTION
**Pull request by tinydylan**
Related Ticket: TINY-6135

Description of Changes:
* Replace some usages of LegacyUnit

Pre-checks:
* [X] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)
* [X] License headers added on new files (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/test/ts/atomic/text/ZwspTest.ts [hunk 1]
```diff
@@ -1,24 +1,8 @@
...
-  });
-
-  suite.test('isZwsp', function () {
-    LegacyUnit.strictEqual(Zwsp.trim('a' + Zwsp.ZWSP + 'b'), 'ab');
-  });
-
-  Pipeline.async({}, suite.toSteps({}), function () {
-    success();
-  }, failure);
+UnitTest.test('ZwspTest', () => {
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/atomic/text/ZwspTest.ts:4]: Please keep the original naming format

File: modules/tinymce/src/core/test/ts/browser/fmt/PreviewTest.ts [hunk 2]
```diff
@@ -1,238 +1,236 @@
...
-import { LegacyUnit, TinyLoader } from '@ephox/mcagar';
-import Editor from 'tinymce/core/api/Editor';
+import { TinyLoader } from '@ephox/mcagar';
 import * as Preview from 'tinymce/core/fmt/Preview';
 import Theme from 'tinymce/themes/silver/Theme';
 import * as HtmlUtils from '../../module/test/HtmlUtils';
 
-UnitTest.asynctest('browser.tinymce.core.fmt.PreviewTest', function (success, failure) {
-  const suite = LegacyUnit.createSuite<Editor>();
+UnitTest.test('Preview.parseSelector()', () => {
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/browser/fmt/PreviewTest.ts:9]: Another one where the prefix in the name shouldn't have been removed

File: modules/tinymce/src/core/test/ts/atomic/text/ZwspTest.ts (hunk 1 above)
Comment (tinydylan) [reply modules/tinymce/src/core/test/ts/atomic/text/ZwspTest.ts]: Fixed. 

Let's discuss test naming in another forum.

File: modules/tinymce/src/core/test/ts/browser/fmt/PreviewTest.ts (hunk 2 above)
Comment (tinydylan) [reply modules/tinymce/src/core/test/ts/browser/fmt/PreviewTest.ts]: Fixed.

**Pull request merged**